unit_test.cpp

```diff
@@ -0,0 +1,121 @@
+#include "Alien.h"
+#include "BaseballPlayer.h"
+#include "BaseballMitt.h"
+#include <unistd.h>
+#include <thread>
+#include <memory>
+#include <mutex>
+#include <iostream>
+#include <assert.h>
+
+#undef NDEBUG
+
+#ifndef NDEBUG
```

Comment:
Well, you don't want this ifndef in your unit test. You just want the ASSERT macro to always be enabled